test/tools/Modules/UnixSocket/UnixSocket.psm1

```diff
@@ -0,0 +1,126 @@
+# Licensed under the MIT License.
+
+Class UnixSocket
+{
+    [System.Management.Automation.Job]$Job
+
+    UnixSocket () { }
+
+    [String] GetStatus()
+    {
+        return $this.Job.JobStateInfo.State
+    }
+}
+
+[UnixSocket]$UnixSocket
+
+function Get-UnixSocket
+{
+    [CmdletBinding(ConfirmImpact = 'Low')]
+    [OutputType([UnixSocket])]
+    param()
+
+    process
+    {
+        return [UnixSocket]$Script:UnixSocket
+    }
+}
+
+function Start-UnixSocket
+{
+    [CmdletBinding(ConfirmImpact = 'Low')]
+    [OutputType([UnixSocket])]
+    param([string] $socketPath)
+
+    process
+    {
+        $runningListener = Get-UnixSocket
+        if ($null -ne $runningListener -and $runningListener.GetStatus() -eq 'Running')
+        {
+            return $runningListener
+        }
+
+        $initTimeoutSeconds  = 25
+        $appExe              = (Get-Command UnixSocket).Path
+        $initCompleteMessage = 'Now listening on'
+        $sleepMilliseconds   = 100
+
+        $Job = Start-Job {
+            $path = Split-Path -Parent (Get-Command UnixSocket).Path -Verbose
+            Push-Location $path -Verbose
+            'appEXE: {0}' -f $using:appExe
+            $env:ASPNETCORE_ENVIRONMENT = 'Development'
+            & $using:appExe $using:socketPath
+        }
+
+        $Script:UnixSocket = [UnixSocket]@{
+            Job   = $Job
+        }
+
+        # Count iterations of $sleepMilliseconds instead of using system time to work around possible CI VM sleep/delays
+        $sleepCountRemaining = $initTimeoutSeconds * 1000 / $sleepMilliseconds
+        do
+        {
+            Start-Sleep -Milliseconds $sleepMilliseconds
+            $initStatus = $Job.ChildJobs[0].Output | Out-String
+            $isRunning = $initStatus -match $initCompleteMessage
+            $sleepCountRemaining--
+        }
+        while (-not $isRunning -and $sleepCountRemaining -gt 0)
+
+        if (-not $isRunning)
+        {
+            $jobErrors = $Job.ChildJobs[0].Error | Out-String
+            $jobOutput =  $Job.ChildJobs[0].Output | Out-String
+            $jobVerbose =  $Job.ChildJobs[0].Verbose | Out-String
+            $Job | Stop-Job
+            $Job | Remove-Job -Force
+            $message = 'UnixSocket did not start before the timeout was reached.{0}Errors:{0}{1}{0}Output:{0}{2}{0}Verbose:{0}{3}' -f ([System.Environment]::NewLine), $jobErrors, $jobOutput, $jobVerbose
+            throw $message
+        }
+        return $Script:UnixSocket
+    }
+}
+
+function Stop-UnixSocket
+{
+    [CmdletBinding(ConfirmImpact = 'Low')]
+    [OutputType([Void])]
+    param()
+
+    process
+    {
+        $Script:UnixSocket.Job | Stop-Job -PassThru | Remove-Job
+        $Script:UnixSocket = $null
+    }
+}
+
+function Get-UnixSocketName {
+    [CmdletBinding()]
+    [OutputType([string])]
+    param ()
+
+    process {
+        $unixSocketName = "UnixSocket.sock"
+        return [System.IO.Path]::Join([System.IO.Path]::GetTempPath(), $unixSocketName)
```

Comment:
```suggestion
        return [System.IO.Path]::Join([System.IO.Path]::GetTempPath(), Path.GetRandomFileName())
```

Unlike [GetTempFileName](https://learn.microsoft.com/en-us/dotnet/api/system.io.path.gettempfilename?view=net-7.0), [GetRandomFileName](https://learn.microsoft.com/en-us/dotnet/api/system.io.path.getrandomfilename?view=net-7.0) does not create a file.

Reply:
I knew about [GetRandomFileName](https://learn.microsoft.com/en-us/dotnet/api/system.io.path.getrandomfilename?view=net-7.0) but I didn't see the advantage of using it

Reply:
@CarloToso Updated this suggestion to keep `sock` extension.

Reply:
`[System.IO.Path]::Join([System.IO.Path]::GetTempPath(), [System.IO.Path]::ChangeExtension([System.IO.Path]::GetRandomFileName(), "sock"))`

Reply:
The sock extension is not required, we can simply use the random name

Reply:
The suggestion was not because the extension is required, but to aid debugging, etc. In any case, it's a minor point.